src/backend/utils/adt/array_userfuncs.c

```diff
@@ -1021,22 +1012,23 @@ array_int4_add(PG_FUNCTION_ARGS)
 	
 	if ( bigenuf1 && bigenuf2 ) /* Conformable arrays. */
 	{
-		Assert(ndatabytes == ndatabytes1 && ndatabytes == ndatabytes2);
+		Assert(ndatabytes == ARR_SIZE(v1) - ARR_DATA_OFFSET(v1) &&
+			   ndatabytes == ARR_DATA_OFFSET(v2));
```

Comment:
Why is it correct to replace `ndatabytes2` with `ARR_DATA_OFFSET(v2)`, but not with `ARR_SIZE(v2) - ARR_DATA_OFFSET(v2)`?